core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.grpc;
+
+import alluxio.collections.Pair;
+import alluxio.conf.AlluxioConfiguration;
+
+import com.google.common.base.MoreObjects;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoopGroup;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Used to identify a unique {@link GrpcChannel}.
+ */
+public class GrpcChannelKey {
+  private static final Random RANDOM = new Random();
+
+  @IdentityField
+  @SuppressFBWarnings(value = "URF_UNREAD_FIELD")
+  Long mPoolKey = 0L;
```

Comment:
Is the pool key only used for equality?

Reply:
Currently `mPoolKey` is a hack for working around pooling for block worker channels. Its value is used for both `hashCode` and `equals`. 
`GrpcChannelKey#setPoolingStrategy` would give you a better idea of its use.